proxy_overrides/base.py

```diff
@@ -13,28 +13,32 @@ def override_model_field(model, name, field):
             model, name, field.__class__.__name__
         ))
 
-    if field.rel:
+    if field.remote_field:
         # Ensure we are referencing another proxy class.
-        if not field.rel.to._meta.proxy:
+        if not field.remote_field.model._meta.proxy:
             raise TypeError('You must relate to another proxy class, not {!r}'.format(
-                field.rel.to.__name__
+                field.remote_field.model.__name__
             ))
 
-        related_name = getattr(field, 'related_name', original_field.related.get_accessor_name())
-        related_model = getattr(field.rel.to, related_name).related.model
+        related_name = field.remote_field.related_name
+        if not related_name:
+            related_name = original_field.remote_field.get_accessor_name()
+        related_model = getattr(field.remote_field.model, related_name).rel.model
+
         if related_model._meta.proxy:
             raise TypeError('There is already a proxy model {!r} related to {!r} using {!r}'.format(
                 related_model.__name__,
-                field.rel.to.__name__,
+                field.remote_field.model.__name__,
                 related_name
             ))
 
     model.add_to_class(name, field)
+    model._meta.local_fields.remove(field)
```

Comment:
Oh, that is sneaky!

Reply:
yes! with this change, the tests run, makemigrations and systems-check does not complain.

the proxyforeignkey code works (which is what I really care about) but proxyfield does not work.

```bash
>tox
GLOB sdist-make: /Users/fgregg/experiment/django-proxy-overrides/setup.py
py37-django20 inst-nodeps: /Users/fgregg/experiment/django-proxy-overrides/.tox/.tmp/package/1/django-proxy-overrides-0.1.0.zip
py37-django20 installed: Django==2.0.9,django-proxy-overrides==0.1.0,pytz==2018.7
py37-django20 run-test-pre: PYTHONHASHSEED='2402096153'
py37-django20 runtests: commands[0] | coverage run -a --branch --source proxy_overrides runtests.py
WARNING: test command found but not installed in testenv
  cmd: /Users/fgregg/experiment/django-proxy-overrides/.env/bin/coverage
  env: /Users/fgregg/experiment/django-proxy-overrides/.tox/py37-django20
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
F.......
======================================================================
FAIL: test_override_compatible (tests.test_field.TestFieldOverride)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fgregg/experiment/django-proxy-overrides/tests/test_field.py", line 15, in test_override_compatible
    self.assertEquals(3, baz_proxy.integer)
AssertionError: 3 != 2

----------------------------------------------------------------------
Ran 8 tests in 0.009s

FAILED (failures=1)
Destroying test database for alias 'default'...
ERROR: InvocationError for command '/Users/fgregg/experiment/django-proxy-overrides/.env/bin/coverage run -a --branch --source proxy_overrides runtests.py' (exited with code 1)
____________________________ summary _____________________________
ERROR:   py37-django20: commands failed
```